components/text.cpp

```diff
@@ -1,26 +1,27 @@
 #include "text.h"
 
 #include "config.h"
+#include "font.h"
 #include "logger.h"
-
-#define STB_TRUETYPE_IMPLEMENTATION
-#include <stb_truetype.h>
+#include "pong.h"
 
 #include <fstream>
 
 namespace pong
 {
 
-Text::Text()
+Text::Text() :
+    mText { "" },
+    mFont { Pong::GetInstance().GetFontBank().GetDefaultFontName() },
```

Comment:
Yeah I see what you mean. I guess I need to understand how realistically testable these classes are to begin with. These text objects will all require a font to already be made in order to do anything useful. But the font already has a lot of various calculations. Is it strange to have a unit test that does a lot of file i/o? Should there be an example .ttf file that is used specfically for unit testing with fonts?

Reply:
Yeah I think that makes sense to have a .ttf to test with specifically.

Note that dependency injection has a bunch of benefits on top of unit testing. It makes the dependency explicit (so future readers know on instantiation, it also forces you to reckon with classes that depend on too many other things because the param list gets nasty). On top of that, you can configure different fonts easily at compile or run time.

Lastly, if you write to an interface, instead of this concrete implementation, you can quickly write new `Font` subclasses really easily without messing with additional code. That's probably overkill for now, but it's good to think about.

All of this boils down to making this class and all supporting classes easier to change in the future. Been listening to a podcast where one of the things that stuck out to me was the quote "every time you change something in your code, note and notice why it was hard to change, how could you have avoided that from the beginning?"

Reply:
Cool youve convinced me!